torch/csrc/distributed/rpc/request_callback_no_python.cpp

```diff
@@ -0,0 +1,625 @@
+    ScriptRemoteCall& scriptRemoteCall,
+    const std::function<void(void)>& postProcessing,
+    std::vector<at::IValue>& stack,
+    const c10::intrusive_ptr<OwnerRRef>& ownerRRef) const {
+  TORCH_CHECK(
+      scriptRemoteCall.hasOp(), "ScriptRemoteCall needs to have an op!");
+  processScriptRemoteCallOp(scriptRemoteCall, postProcessing, stack, ownerRRef);
+}
+
+bool RequestCallbackNoPython::processScriptRemoteCallOp(
+    ScriptRemoteCall& scriptRemoteCall,
+    const std::function<void(void)>& postProcessing,
+    std::vector<at::IValue>& stack,
+    const c10::intrusive_ptr<OwnerRRef>& ownerRRef) const {
+  if (scriptRemoteCall.hasOp()) {
+    try {
+      scriptRemoteCall.op()->getOperation()(&stack);
+    } catch (const std::exception& e) {
+      // Don't throw in this call, but rather transfer the exception
+      // to the rref.
+      ownerRRef->setError(e.what());
+      postProcessing();
+      return true;
+    }
+    TORCH_INTERNAL_ASSERT(
+        stack.size() == 1,
+        "Return value of a builtin operator or a "
+        "TorchScript function should be a single IValue, got a vector of "
+        "size ",
+        stack.size());
+    ownerRRef->setValue(std::move(stack.front()));
+    postProcessing();
+    return true;
+  }
+  return false;
+}
+
+void RequestCallbackNoPython::processPythonRRefFetchCall(
+    RpcCommandBase& rpc,
+    const int64_t messageId,
+    const std::shared_ptr<FutureMessage>& responseFuture) const {
+  C10_THROW_ERROR(Error, "Python call not supported!");
+}
+
+void RequestCallbackNoPython::handleRRefDelete(
+    c10::intrusive_ptr<RRef>& rref) const {
+  TORCH_CHECK(!rref->isPyObj(), "RRefs with python objects not supported!");
+}
+
+void RequestCallbackNoPython::processRpc(
+    RpcCommandBase& rpc,
+    const MessageType& messageType,
+    const int64_t messageId,
+    const std::shared_ptr<FutureMessage>& responseFuture) const {
+  auto markComplete = [messageId, &responseFuture](Message m) {
+    m.setId(messageId);
+    responseFuture->markCompleted(std::move(m));
+  };
+
+  // TODO: RpcCommandBase should have an abstract execute() method that we can
+  // call here instead of having another switch statement here. Even better we
+  // could have abstract classes RpcRequest and RpcResp which inherit from
+  // RpcCommandBase and RpcRequest declares the abstract method execute() that
+  // we can call here. RpcResponse could have an abstract method to convert it
+  // to a python object.
+  switch (messageType) {
+    case MessageType::SCRIPT_CALL: {
+      auto& scriptCall = static_cast<ScriptCall&>(rpc);
+
+      // scriptCall is only alive within this block, use reference to avoid copy
+      auto& stack = scriptCall.stackRef();
+      processScriptCall(
+          scriptCall, markComplete, stack, messageId, responseFuture);
+      return;
+    }
+    case MessageType::PYTHON_CALL: {
+      processPythonCall(rpc, markComplete, messageId, responseFuture);
```

Comment:
any reason for abstract some code into a dedicated method, but leaving others out (e.g., `SCRIPT_REMOTE_CALL`)?

Reply:
I mostly focussed on refactoring the pieces that enable the python vs no python version. Didn't want to pile on too many more changes for this PR.